packages/publint/src/index.d.ts

```diff
@@ -120,6 +120,7 @@ export type Message =
       }
     >
   | BaseMessage<'LOCAL_DEPENDENCY'>
+  | BaseMessage<'BIN_FILE_MISSING_NODE_SHEBANG'>
```

Comment:
I can drop the `BIN_` prefix to make it more generic if that's preferred.

Reply:
I think having the BIN_ is great for now. I'm not sure if there's any other cases where we'd check the shebang again in the future.

I've been thinking about the checks, and I'm a bit worried that it can catch false positives like:

- Bash files that are executable itself
- Bash files with `#!/usr/bin/env bash`
- Js files that choose to support only a specific runtime like `#!/usr/bin/env bun`

They're not very common setups, but i think we can avoid flagging them if possible. Maybe we can tone it down to only check against JS files, and that it starts with `#!/usr/bin/env` only? And rename the rule as `BIN_FILE_NOT_EXECUTABLE`?

Reply:
I've made the following updates
- Relaxed the checks to only run against JS files.
- Only check that the file starts with `#!/usr/bin/env`.
- Renamed the rule to `BIN_FILE_NOT_EXECUTABLE`.